### PR TITLE
add normalize, range, scale_each to make_grid

### LINKED ISSUE #24: save_image behavior
https://github.com/pytorch/vision/blob/master/torchvision/utils.py#L52

The `.mul(0.5).add(0.5)` in this line seems to be assuming that the input values are in [-1,1]? Isn't the Torch convention to use [0,1] for images?

### PR DESCRIPTION
**Pull request by soumith**
fixes #24 

cc: @rtqichen @alykhantejani @gpleiss does this look better?

### CODE REVIEW

**Comment by soumith** [comment]
also cc: @fmassa for code review :)

File: torchvision/utils.py [hunk 1]
```diff
@@ -42,14 +42,19 @@ def make_grid(tensor, nrow=8, padding=2):
...
     """
     Saves a given Tensor into an image file.
     If given a mini-batch tensor, will save the tensor as a grid of images.
     """
     from PIL import Image
     tensor = tensor.cpu()
     grid = make_grid(tensor, nrow=nrow, padding=padding)
-    ndarr = grid.mul(255).byte().transpose(0,2).transpose(0,1).numpy()
+    if range is not None:
+        assert isinstance(tuple, range), "range has to be a tuple (min, max) if specified"
```
Comment (fmassa) [inline torchvision/utils.py]: I think the arguments in here are inverted, and should be instead `isinstance(range, tuple)`

File: torchvision/utils.py [hunk 2]
```diff
@@ -42,14 +42,19 @@ def make_grid(tensor, nrow=8, padding=2):
...
+        grid = torch.clamp(grid, min=range[0], max=range[1])
+        # make grid to be [0, 1]
+        grid.add_(-range[0]).div(range[1] - range[0])
```
Comment (fmassa) [inline torchvision/utils.py]: I think this should be an in-place division.

Comment (alykhantejani) [inline torchvision/utils.py]: I think you need `div_` her

Comment (soumith) [inline torchvision/utils.py]: this is what happens when i make prs in a hurry. had to run out. will test pr tomo, fix it and merge.

**Comment by fmassa** [comment]
@soumith apart from the two inline comments, seems good to me!

File: torchvision/utils.py (hunk 2 above)
Comment (soumith) [inline torchvision/utils.py]: I was trying to be a ninja

**Comment by alykhantejani** [comment]
@soumith LGTM except for the 2 comments

**Comment by soumith** [comment]
Made this PR more complete.
Add three options:
`normalize`, `range` and `scale_each`.
Here is a notebook showcasing them:
https://gist.github.com/anonymous/bf16430f7750c023141c562f3e9f2a91

screen shot 2017-03-15 at 8 36 24 pm

**Comment by fmassa** [comment]
Look very nice, thanks Soumith!
